### PR TITLE
Remove client from _ContainerApp and _LocalApp

### PR DESCRIPTION
**Pull request by erikbern**
Moving it to the stub instead.

More prep work for turning stubs into apps.

### CODE REVIEW

File: modal/stub.py [hunk 1]
```diff
@@ -301,11 +303,13 @@ def is_inside(self, image: Optional[_Image] = None):
...
 
     @asynccontextmanager
-    async def _set_local_app(self, app: _LocalApp) -> AsyncGenerator[None, None]:
+    async def _set_local_app(self, client: _Client, app: _LocalApp) -> AsyncGenerator[None, None]:
+        self._client = client
         self._local_app = app
         try:
             yield
         finally:
+            self._client = client
```
[the remark this answers is not in the document]
Comment (mwaskom) [reply modal/stub.py:312]: Or should we be cleaning it up by setting to `None`?

[the remark this answers is not in the document]
Comment (erikbern) [reply modal/stub.py:312]: oops sorry this should be `None`

thanks for catching!

**Comment by erikbern** [comment]
@prbot approve

**Review by mwaskom — APPROVED** [review]
:+1:

Not sure why @prbot tagged @gongy. Bad bot!

**Pull request merged**